# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/tsa/slrn/slrn-0.9.8.1-fetch.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- slrn-0.9.8.1/src/slrnpull.c	2004-02-28 23:08:17.000000000 +0100
+++ slrn/src/slrnpull.c	2004-10-10 01:42:31.000000000 +0200
@@ -1647,7 +1647,7 @@
 	    || (((unsigned int)server_max <= gmax) && (gmin <= gmax)))
 	  {
 	     log_message (_("%s: no new articles available."), g->name);
-	     return 0;
+	     goto fetch_marked_bodies;
 	  }
 	
 	Num_Duplicates = 0;
@@ -1667,7 +1667,7 @@
 	     g->max = g->server_max;
 	     log_message (_("%s: No new articles available."), g->name);
 	     slrn_free ((char *) numbers);
-	     return 0;
+	     goto fetch_marked_bodies;
 	  }
 	
 	log_message (_("%s: %u articles available."), g->name, num_numbers - i);
@@ -1716,7 +1716,8 @@
      }
 
    /* Now, fetch marked article bodies. */
-   
+
+   fetch_marked_bodies:
    /* Don't request bodies that are no longer there. */
    if (server_min > 1)
      g->requests = slrn_ranges_remove (g->requests, 1, server_min-1);
